src/aqua/installation-spell/src/aqua/deploy.aqua

```diff
@@ -75,6 +75,21 @@ func deploy(config: AppDeployDefinition) -> DeployedAppWorkers:
     deployed = DeployedAppWorkers(workers = workers)
     <- deployed
 
+data RemovedWorker:
+    name: string
+    worker_ids: []string
 
--- TODO:
--- func remove(worker: WorkerId)
\ No newline at end of file
+-- remove deployed workers
+func remove(workers: DeployedAppWorkers) -> []RemovedWorker:
+    removed_workers: *RemovedWorker
+    for w <- workers.workers:
+        removed_worker_ids: *string
+        for h <- w.installation_spells:
+            on h.host_id:
+                Worker.remove(h.worker_id)
```

Comment:
```suggestion
                -- This will raise an error if INIT_PEER_ID is different from Worker's owner
                -- That's because only the one who deployed the worker, can remove it
                Worker.remove(h.worker_id)
```